src/common/snippets/src/lowered/pass/insert_reg_spills.cpp

```diff
@@ -0,0 +1,88 @@
+            start_it--;
+            stop_it++;
+        }
+        // Note: we need to insert immediately before LoopBegin => increment start_it
+        start_it++;
+        const auto& loop_begin_live = start_it->get()->get_live_regs();
+        std::set<Reg> used;
+        const auto& reg_info = expr->get_reg_info();
+        used.insert(reg_info.first.begin(), reg_info.first.end());
+        used.insert(reg_info.second.begin(), reg_info.second.end());
+        // Note: before the loop, we need to spill all live regs except for the ones used by the target expression
+        std::set<Reg> regs_to_spill;
+        std::set_difference(loop_begin_live.begin(), loop_begin_live.end(),
+                            used.begin(), used.end(),
+                            std::inserter(regs_to_spill, regs_to_spill.begin()));
+        // we also need to keep kernel regs alive (actually only abi_param_1 is used in emitters, but save all for consistency)
+        for (const auto& r : m_reg_manager.get_kernel_call_regs( snippets::op::Kernel::make_kernel(linear_ir.is_dynamic())))
+            regs_to_spill.erase(r);
+        if (regs_to_spill.empty())
+            continue;
+        // All spilled regs are not live anymore => update live_regs for affected expressions
+        for (auto affected_it = start_it; affected_it != stop_it; affected_it++) {
+            const auto& affected_expr = *affected_it;
+            const auto& live_old = affected_expr->get_live_regs();
+            std::set<Reg> live_new;
+            std::set_difference(live_old.begin(), live_old.end(),
+                                regs_to_spill.begin(), regs_to_spill.end(),
+                                std::inserter(live_new, live_new.begin()));
+            affected_expr->set_live_regs(live_new);
+        }
+
+        const auto begin = std::make_shared<op::RegSpillBegin>(regs_to_spill);
+        const auto end = std::make_shared<op::RegSpillEnd>(begin);
+        const auto loop_ids = start_it->get()->get_loop_ids();
+        OPENVINO_ASSERT(loop_ids == std::prev(stop_it)->get()->get_loop_ids(), "Inconsistent loop ids for RegSpill expressions");
+        const auto spill_begin_expr = *linear_ir.insert_node(begin, std::vector<PortConnectorPtr>{}, loop_ids,
+                                                             false, start_it, std::vector<std::set<ExpressionPort>>{});
+        std::vector<Reg> vregs{regs_to_spill.begin(), regs_to_spill.end()};
+        spill_begin_expr->set_reg_info({{}, vregs});
+        spill_begin_expr->set_live_regs(std::prev(start_it, 2)->get()->get_live_regs());
+
+        const auto spill_end_expr = *linear_ir.insert_node(end, spill_begin_expr->get_output_port_connectors(), loop_ids,
+                                                           false, stop_it, std::vector<std::set<ExpressionPort>>{});
+        spill_end_expr->set_reg_info({vregs, {}});
+        spill_begin_expr->set_live_regs(stop_it->get()->get_live_regs());
```

Comment:
Could you please add a comment what the live regs we set to `spill_begin_expr` and `spill_end_expr`? `stop_it` is the next expr after `LoopEnd`: this is quite obvious. `std::prev(start_it, 2)` is the expr before `LoopBegin`, am I right? But this is not easy to understand, especially if forget that `start_it` is separately incremented on L38

Reply:
Now all is totally clear, thanks for the refactoring!